ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -629,17 +621,13 @@ let clean_local_resources () : unit =
     error "%s : %s" msg (ExnHelper.string_of_exn e) ;
     raise (Auth_service_error (E_GENERIC, msg))
 
-let domainify_uname ~domain uname =
-  let open Astring.String in
-  if
-    is_infix ~affix:domain uname
-    || is_infix ~affix:"@" uname
-    || is_infix ~affix:{|\|} uname
-    || uname = krbtgt
-  then
-    uname
-  else
-    Printf.sprintf "%s@%s" uname domain
+let undomainify_uname uname =
```

Comment:
Would it be easier to extract the desired substring using a regular expression and do this with a single operation? I don't like the piecemeal destruction of strings because it tends to be unreliable.